parboiled-core/src/test/scala/org/parboiled2/ActionSpec.scala

```diff
@@ -147,10 +146,10 @@ object ActionSpec extends TestParserSpec {
 
       case class Foo(i: Int, s: String)
 
-      /*FIXME: type error "`~>` producing case class (simple notation)" - new TestParser1[Foo] {
-        def targetRule = rule(push(1 :: "X" :: HNil) ~> Foo)
+      "`~>` producing case class (simple notation)" - new TestParser1[Foo] {
+        def targetRule = rule(push(1 :: "X" :: HNil) ~> Foo.apply)
```

Comment:
`Foo.apply` itself won't work in Scala 2. For compatibility you'll need `Foo.apply _` or `Foo(_, _)`

Reply:
https://dotty.epfl.ch/docs/reference/changed-features/eta-expansion-spec.html
```
In Scala 2, a method reference m is converted to a function value only if the expected type is a function type
```